src/main/java/com/limechain/sync/warpsync/WarpSyncState.java

```diff
@@ -281,8 +270,13 @@ public void syncNeighbourMessage(NeighbourMessage neighbourMessage, PeerId peerI
     }
 
     private void updateSetData(BigInteger setChangeBlock) {
+
         List<BlockData> response = requester.requestBlockData(
-                BlockRequestField.ALL, setChangeBlock.intValue(), 1).join();
+                BlockRequestField.ALL,
+                setChangeBlock.intValue(),
```

Comment:
I know you haven't written this but NEVER use `intValue()` over `intValueExact()` - it doesn't check for overflows and always returns a result. If you see it elsewhere in the code change it there as well.